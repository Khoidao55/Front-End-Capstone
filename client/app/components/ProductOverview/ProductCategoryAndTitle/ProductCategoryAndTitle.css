.product-star {
  display: flex;
}

.read-all-review {
  text-decoration: underline;
  margin-left: 10px;
}

.read-all-review:hover{
  cursor: pointer;
  color: blue;
}