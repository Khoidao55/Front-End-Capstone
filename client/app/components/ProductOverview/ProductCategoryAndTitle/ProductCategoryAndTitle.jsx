import React from 'react';
import './ProductCategoryAndTitle.css';
import StarRating from '../../StarRating/StarRating.jsx';

const ProductCategoryAndTitle = ({ reviewNumber, price, product }) => {
  if (product !== undefined && reviewNumber?.results !== undefined) {
    const clickedReviewFunc = (e) => {
      e.preventDefault();
      return document.querySelector('#review-list').scrollIntoView();
    };

    return (
      <div className="categoryAndTitleContainer">
        <div className={`product-star-${reviewNumber.results.length !== 0 ? 'active' : 'disabled'}`}>
          <StarRating reviewScore={product.averageRating} setMargin='16px 0 0 0' />
          <p onClick={(e) => clickedReviewFunc(e)}className='read-all-review'>{`read (${reviewNumber.results.length}) reviews`}</p>
        </div>
        <p>{product.category}</p>
        <h1>{product.name}</h1>
        <p>${price}</p>
      </div>
    );
  }

  return (
    <></>
  );
};

export default ProductCategoryAndTitle;
