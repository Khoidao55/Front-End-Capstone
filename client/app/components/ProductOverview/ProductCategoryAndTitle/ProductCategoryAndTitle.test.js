import React from 'react';
import { shallow } from 'enzyme';
import ProductCategoryAndTitle from './ProductCategoryAndTitle.jsx';

const dummyProps = {
  product: {
    id: 12345,
    productImage: 'img',
    category: 'shirt',
    name: 'cool shirt',
    originalPrice: '123',
    salePrice: '123',
    averageRating: 5,
  },
};

describe('Product Category and Title', () => {
  test('should render my component', () => {
    const wrapper = shallow(<ProductCategoryAndTitle {...dummyProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
