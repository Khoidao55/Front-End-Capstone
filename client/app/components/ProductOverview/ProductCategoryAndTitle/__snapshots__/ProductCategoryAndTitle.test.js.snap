// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Category and Title should render my component 1`] = `ShallowWrapper {}`;
